### ISSUE TITLE
Clustering microservices

### ISSUE THREAD

**Issue by daffl** [feathersjs/feathers#121]
## 1) Event propagation in a cluster

When two instance of the same application are running in different processes or on different machines `created`, `updated`, `patched` and `removed` service events should be propagated through the entire cluster. That way every connected client will also be notified of any change no matter where it happened.

This can be done by using a master (e.g. the [cluster](https://iojs.org/api/cluster.html) master) as the event dispatching hub and have all client services subscribe and emit events to that master event emitter which then dispatches it to all slaves. It might be necessary for different implementations e.g. the Cluster module and Redis (or other message queuing mechanisms).

## 2) Individual remote services

The other case is having different services distributed over different applications. Once #118 is resolved this should be simply possible by dropping in a remote service using any communication mechanism (e.g. websockets) like:

```js
// todos.js
var feathers = require('feathers');
var app = feathers();
var wsClient = require('feathers-websocket-client');

app.use('/todos', todoService);
app.use('/users', wsClient('http://api.myapp.com/users'));
```

**Comment by daffl** [comment]
I created [feathers-mubsub](https://github.com/feathersjs/feathers-mubsub) for 1) which does event synchronization via MongoDB publish/subscribe.

**Comment by daffl** [comment]
2) Can be solved with [feathers-client](https://github.com/feathersjs/feathers-client/) and #136 so this issue can be closed in 1.1.

**Comment by daffl** [comment]
Closing this issue for now. Both points can be solved with existing 1.1 functionality and plugins.
